tests/util/test_math.py

```diff
@@ -0,0 +1,49 @@
+import scipy
+import scipy.integrate
+from scipy.stats import norm
+from numpy import log, sqrt, exp, inf
+
+from util import test_statistic_threshold as tst
+
+
+def kullback_leibler(post_sample, post_pdf, comp_pdf):
+    """Estimate KL-divergence of sample (a collection of values) w.r.t. known pdf,
+    comp_pdf, which returns the density when passed a sample. Return value is
+    (estimated_kl, variance) post_sample is an approximate sample from the
+    distribution approximately represented by post_pdf
+
+    """
+    klsamples = [log(post_pdf(x)) - log(comp_pdf(x)) for x in post_sample]
+    klsamples = scipy.array(klsamples)
+    # In general the mean and variance of log(P(x)) is not known to be finite,
+    # but it will be for any distribution crosscat generates at the moment,
+    # because they all have finite entropy.  Hence CLT applies
+    return klsamples.mean(), klsamples.var() / sqrt(len(klsamples))
```

Comment:
This doesn't look numerically reliable. Consider computing entirely in the log domain.